command/snapshot/inspect/snapshot_inspect.go

```diff
@@ -175,16 +218,19 @@ func (r *countingReader) Read(p []byte) (n int, err error) {
 
 // enhance utilizes ReadSnapshot to populate the struct with
 // all of the snapshot's itemized data
-func enhance(file io.Reader) (map[structs.MessageType]typeStats, int, error) {
+func enhance(file io.Reader, detailed bool, depth int, filter string) (map[structs.MessageType]typeStats, map[string]typeStats, int, int, error) {
```

Comment:
Could we tie this to `(c *cmd)` so those three flags won't have to be passed around, or is there a way for us to not need to tie into `*cmd` at all? 
Either ways are fine :)